src/main/java/com/couchbase/lite/LiveQuery.java

```diff
@@ -149,93 +117,87 @@ ListenerToken addChangeListener(Executor executor, QueryChangeListener listener)
      * NOTE: this method is synchronized with Query level.
      */
     void removeChangeListener(ListenerToken token) {
+        if (changeNotifier.removeChangeListener(token) <= 0) { stop(); }
+    }
+
+    /**
+     * Starts observing database changes and reports changes in the query result.
+     */
+    void start() {
+        final Database db = query.getDatabase();
+        if (db == null) { throw new IllegalArgumentException("live query database cannot be null."); }
+
         synchronized (lock) {
-            if (changeNotifier.removeChangeListener(token) == 0) { stop(true); }
+            if (state.compareAndSet(State.STOPPED, State.STARTED)) {
+                db.addActiveLiveQuery(this);
+                dbListenerToken = db.addChangeListener(this);
+            }
+
+            // Here if state is STARTED or SCHEDULED:
+            // this method may be called while the LiveQuery is already running,
+            // for example, by Query.setParameters().
+            // If that happens we want to kick off another query, right now
+            releaseResultSetSynchronized();
```

Comment:
If adding a new listener, the current query result set will get released as well. The behavior would be:
1. If the live query is not started (no current result set), the query will be executed.
2. If the query parameter is changed, the current result set will be released and the query will be re-executed.
3. If a new listener is added and there is a current result set, the new added listener should be notified with the current result set (I think CBL iOS has a bug not doing this).

Reply:
Oh!!  Good catch!!!  Thank you, Pasin!